Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -503,8 +525,11 @@ public void Run()
                 _algorithm.RunTimeError = err;
             }
 
-            Log.Trace("BrokerageTransactionHandler.Run(): Ending Thread...");
-            IsActive = false;
+            if (_processingThread != null)
+            {
+                Log.Trace("BrokerageTransactionHandler.Run(): Ending Thread...");
+                IsActive = false;
```

Comment:
Do we keep `IsActive = false` for the non-threaded version? Perhaps the bigger question is what does `IsActive` imply? Does it imply that it can handle requests (contract) or does it imply a thread is running (implementation) -- I try to stick to contractual definitions vs implementation definitions as they're usually more geared towards the desired behavior vs how that behavior was implemented.